perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupAndGroupStructureSynchronizationIntegrationTest.java

```diff
@@ -1088,6 +1093,239 @@ public void synchronizeGroupUpdateExistingMemberStatus() throws Exception {
+		String A_MG_D_MEMBERSHIP_EXPIRATION = AttributesManager.NS_MEMBER_GROUP_ATTR_DEF + ":groupMembershipExpiration";
+
+		when(extSourceManagerBl.getExtSourceByName(sess, ExtSourcesManager.EXTSOURCE_NAME_PERUN)).thenReturn(extSourceForUserCreation);
+
+		Attribute attr = attributesManagerBl.getAttribute(sess, group, GroupsManager.GROUPEXTSOURCE_ATTRNAME);
+		attr.setValue(extSource.getName());
+		attributesManagerBl.setAttribute(sess, group, attr);
+
+		List<Map<String, String>> subjects = new ArrayList<>();
+		Map<String, String> attributes = new HashMap<>();
+		attributes.put("login", "metodej");
+		attributes.put("status", "EXPIRED");
+		subjects.add(attributes);
+		Candidate candidate = setUpCandidate();
+		candidate.setExpectedSyncGroupStatus("EXPIRED");
+
+		member = perun.getMembersManagerBl().createMemberSync(sess, vo, candidate);
+		groupsManagerBl.addMember(sess, group, member);
+
+		when(extSourceManagerBl.getCandidate(sess, attributes, (ExtSourceLdap)essa, "metodej")).thenReturn(new CandidateSync(candidate));
+		when(essa.getGroupSubjects(anyMap())).thenReturn(subjects);
+
+		groupsManagerBl.synchronizeGroup(sess, group);
+		assertEquals(MemberGroupStatus.EXPIRED, groupsManagerBl.getGroupMembers(sess, group).get(0).getGroupStatus());
+		assertEquals(LocalDate.now().toString(), attributesManagerBl.getAttribute(sess, member, group, A_MG_D_MEMBERSHIP_EXPIRATION).getValue());
+	}
+
+	@Test
+	public void synchronizeExpiredGroupStatusMemberNotInVO() throws Exception {
+		System.out.println(CLASS_NAME + "synchronizeGroupUpdateExistingMemberStatus");
+
+		when(extSourceManagerBl.getExtSourceByName(sess, ExtSourcesManager.EXTSOURCE_NAME_PERUN)).thenReturn(extSourceForUserCreation);
+
+		Attribute attr = attributesManagerBl.getAttribute(sess, group, GroupsManager.GROUPEXTSOURCE_ATTRNAME);
+		attr.setValue(extSource.getName());
+		attributesManagerBl.setAttribute(sess, group, attr);
+
+		List<Map<String, String>> subjects = new ArrayList<>();
+		Map<String, String> attributes = new HashMap<>();
+		attributes.put("login", "metodej");
+		attributes.put("status", "EXPIRED");
+		subjects.add(attributes);
+		Candidate candidate = setUpCandidate();
+		candidate.setExpectedSyncGroupStatus("EXPIRED");
+
+		when(extSourceManagerBl.getCandidate(sess, attributes, (ExtSourceLdap)essa, "metodej")).thenReturn(new CandidateSync(candidate));
+		when(essa.getGroupSubjects(anyMap())).thenReturn(subjects);
+
+		groupsManagerBl.synchronizeGroup(sess, group);
+		assertEquals(MemberGroupStatus.EXPIRED, groupsManagerBl.getGroupMembers(sess, group).get(0).getGroupStatus());
+	}
+
+	@Test
+	public void synchronizeGroupUpdateMembershipStatusToDefaultValid() throws Exception {
+		System.out.println(CLASS_NAME + "synchronizeGroupUpdateMembershipStatus");
+
+		Group groupWithExpiration = new Group("GroupsManagerTestGroup2","testovaci2");
+		groupWithExpiration = groupsManagerBl.createGroup(sess, vo, groupWithExpiration);
```

Comment:
Please create a private method to setup `groupWithExpiration` since it repeats itself in most of the tests